src/remora/data_chunks.py

```diff
@@ -98,11 +99,28 @@ def make_sequence_coordinate_mapping(cigar):
     query_knots = np.concatenate(
         [[0], (query_knots[is_match] - offsets).T.flatten(), [query_knots[-1]]]
     )
-    knots = np.interp(np.arange(ref_knots[-1] + 1), ref_knots, query_knots)
+    ref_coords = get_ref_coords(ops, lens)
+    knots = np.interp(np.concatenate([[0], ref_coords]), ref_knots, query_knots)
 
     return knots
 
 
+def get_ref_coords(ops, lens):
+    '''
+    Map query base to reference coordinates (1-based). 
+    Returns 
+        array shape (ref_len,); ref_len = len(aln.get_reference_sequences())
+    '''
+    starts = np.cumsum(np.where(REF_OPS[ops], lens, 0)) - np.where(REF_OPS[ops], lens, 0)
+    ranges = [
+        np.arange(start, start + l)
+        for op, start, l in zip(ops, starts, lens)
+        if REFCOORD_OPS[op]
+    ]
+    ref_coords = np.concatenate(ranges) + 1
```

Comment:
I still feel like there is extra logic here. I think the fix here might be as simple as completely dropping the skip cigar ops. Do you get the same results if you switch the `REF_OPS` skip code value to `False`?

Reply:
I was thinking about the same thing I think you're right - just changing the intron operation to False in REF_OPS can work as well... got the same results for genomic reads and didn't raise errors for transcriptomic reads